plugins/upower/fu-plugin-upower.c

```diff
@@ -46,6 +48,42 @@ fu_plugin_startup (FuPlugin *plugin, GError **error)
 	return TRUE;
 }
 
+static gboolean
+upower_charged_enough (void)
+{
+	gdouble level;
+	g_autoptr(GVariant) val = NULL;
+	g_autoptr(GError) error_local = NULL;
+	g_autoptr(GDBusProxy) proxy =
+		g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
+					       G_DBUS_PROXY_FLAGS_NONE,
+					       NULL,
+					       "org.freedesktop.UPower",
+					       "/org/freedesktop/UPower/devices/DisplayDevice",
+					       "org.freedesktop.UPower.Device",
+					       NULL,
+					       &error_local);
+	if (proxy == NULL) {
+		g_printerr ("%s", error_local->message);
+		return FALSE;
```

Comment:
upower has a `DaemonVersion` attribute.  What do you think about just check that at plugin startup instead and if it's too old then reject plugin startup alltogether rather than these logical fallacies?

Reply:
We could, although I don't think it buys us much. I think we want to default to allowing flashing for "prepare" check like this.

Reply:
OK, understand.